devlib/trace/ftrace.py

```diff
@@ -225,6 +225,9 @@ def reset(self):
             self._set_buffer_size()
         self.target.execute('{} reset'.format(self.target_binary),
                             as_root=True, timeout=TIMEOUT)
+        if self.functions:
```

Comment:
Might also check that `self.tracer` is `None`.

Reply:
I don't know about `None`, but it should at least be checked against `function_graph` since that is the goal of the patch.

Reply:
I'm just thinking if there would be cases that do not have this exposed and still want to use `function_graph`. 
In which case maybe change this echo to `self.target.write_value(self.function_profile_file, 0, verify=False)` so that the value is written if it can be and if not, any errors can be safely ignored as they should be picked up elsewhere in the code that was actively using it?